lib/versioning/gradle/index.spec.ts

```diff
@@ -113,6 +118,16 @@ describe('versioning/gradle/index', () => {
     expect(range).toBeNull();
   });
 
+  test.each`
+    version      | expected
+    ${'1'}       | ${[{ type: 1, val: 1 }]}
+    ${undefined} | ${null}
+    ${'latest'}  | ${null}
+    ${'1!a!1'}   | ${null}
+  `('valid("$input") === $expected', ({ version, expected }) => {
+    expect(JSON.stringify(valid(version))).toBe(JSON.stringify(expected));
+  });
+
```

Comment:
Not required if you update `isValid` 😉

Reply:
But `isValid` needs to return `boolean` as per `VersioningApi` so it's better to use `valid` to prevent confusion.

Reply:
It's okay to not explicit Test every small function. I think `isValid` tests should cover all code paths when updated.